caffe2/proto/caffe2.proto

```diff
@@ -86,7 +126,13 @@ message TensorShape {
   repeated int32 unknown_dims = 3;
   optional bool unknown_shape = 4 [default = false];
   optional string name = 5;
+}
 
+// This is prepared for non-tensor types.
+// Right now, it only contains name, later we may add some native type
+// support in proto like ONNX.
+message SpecialType {
```

Comment:
ok, so the plan is to get rid of the BlobProto and use TensorProto for it?

Reply:
I am OK both ways but please make sure that the corresponding plumbing - like Blob serialization etc, are properly routed.

Reply:
Yes, get rid of BlobProto and use TensorProto to replace it